lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -632,17 +707,35 @@ instance
     ( SoftDerivation k
     , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
     , MkKeyFingerprint k Address
-    ) => IsOurs (SeqState n k) Address where
-    isOurs addr (SeqState !s1 !s2 !ixs !rpk) =
+    ) => IsOurs (SeqState n k) Address
+  where
+    type DerivationPath (SeqState n k) Address =
+        ( Index 'Hardened 'PurposeK
+        , Index 'Hardened 'CoinTypeK
+        , Index 'Hardened 'AccountK
+        , Index 'Soft 'RoleK
+        , Index 'Soft 'AddressK
+        )
+    isOurs addr (SeqState !s1 !s2 !ixs !rpk !prefix) =
         let
+            DerivationPrefix (purpose, coinType, accountIx) = prefix
             (internal, !s1') = lookupAddress @n (const Used) addr s1
             (external, !s2') = lookupAddress @n (const Used) addr s2
+
             !ixs' = case internal of
                 Nothing -> ixs
                 Just ix -> updatePendingIxs ix ixs
-            ours = isJust (internal <|> external)
+
+            ours = case (external, internal) of
+                (Just addrIx, _) ->
+                    Just (purpose, coinType, accountIx, utxoExternal, addrIx)
+
+                (_, Just addrIx) ->
+                    Just (purpose, coinType, accountIx, utxoInternal, addrIx)
+
+                _ -> Nothing
```

Comment:
The other main change is here, in the definition of `isOurs`. Instead of returning a mere `Bool`, we now return the full derivation path corresponding to that address.